### ISSUE TITLE
Consider making container cli configurable?

### ISSUE THREAD

**Issue by jchorl** [bazel-contrib/rules_oci#430]
Hi!

First, thanks for rules_oci. It's great.

I noticed that `command -v docker` and `command -v podman` are hardcoded in https://github.com/bazel-contrib/rules_oci/blob/375e5dbf3b2c50e83c9df0d0dc16a027984dab77/oci/private/tarball_run.sh.tpl

Notably, this assumes your local env has docker/podman cli executables. `rules_docker` had [`tool_target`](https://github.com/bazelbuild/rules_docker/blob/3040e1fd74659a52d1cdaff81359f57ee0e2bb41/toolchains/docker/toolchain.bzl#L98), which enabled specifying a target for a docker executable.

It would be nice if you could provide a docker bin to rules_oci to use.

"Why would you ever want to load things into local docker, without having the docker executable available?"

The [bazel docker container](https://bazel.build/install/docker-container) doesn't have the docker binary. It's pretty trivial to mount the docker socket in there. In theory you could also mount your docker binary to `/usr/bin/docker`, but that feels unnecessary.

There's a lot of prebuilt docker cli static binaries floating around. It would be pretty easy to just download one and inject it into `rules_oci` to use via target.

What do you think? Is this a PR/feature you would entertain?

**Comment by thesayyn** [comment]
https://github.com/bazel-contrib/rules_oci#design states that we are not docker or any other runtime specific. we only conform to oci spec.  

TLDR; We will never fetch anything runtime/brand-specific binaries here.

the `bazel run` feature of oci_tarball is just for convenience, and we don't plan to expand it. it's not hermetic by nature of bazel run and therefore never guaranteed to work.

**Comment by jchorl** [comment]
Thanks for the response. Sorry, I think my ask may have been unclear.

> It would be nice if you could provide a docker bin to rules_oci to use.

I meant, it would be nice if _I_ (the user) could pass in a target that `oci_tarball` would use as its command (instead of `command -v`). In that case, the tool wouldn't pull a docker bin, but I could certainly pull a docker/podman bin and pass it to the tool.

Would accepting a label to configure the toolchain be in-scope? If not, no worries, I can understand why you might want to limit complexity.

I'm envisioning something like (in WORKSPACE):
```
oci_register_toolchains(
    name = "oci",
    crane_version = LATEST_CRANE_VERSION,
    container_cli = "//my/cli:bin",
)
```

**Comment by thesayyn** [comment]
is https://github.com/bazel-contrib/rules_oci/issues/295 what you are looking for? it's less likely we'll make this configurable with a toolchain. easiest solution is that you provide us a binary in oci_tarball and we run that instead of docker|podman.

**Comment by jchorl** [comment]
> is #295 what you are looking for? it's less likely we'll make this configurable with a toolchain. easiest solution is that you provide us a binary in oci_tarball and we run that instead of docker|podman.

Yep - that's exactly what I was looking for. I'll follow that issue. Thanks!
